artifact-manifest-spec.md

```diff
@@ -0,0 +1,76 @@
+# OCI Artifact Manifest Spec (Phase-1 Reference Types)
+The OCI artifact manifest generalizes the use cases of [OCI image manifest][oci-image-manifest-spec] by removing constraints defined on the image-manifest such as a required `config` object and required & ordinal `layers`. It then adds a `subjectManifest` property supporting reference types. The addition of a new manifest does not change, nor impact the `image.manifest`. It provides a means to define a wide range of artifacts, including a chain of related artifacts enabling SBoMs, on-demand loading, signatures and metadata that can be related to an `image.manifest` or `image.index`. By defining a new manifest, registries and clients opt-into new capabilities, without breaking existing registry and client behavior or setting expectations for scenarios to function when the client and/or registry doesn't yet implement the new capabilities.
+
+To enable a fall 2021 focus on supply chain security,  **Phase 1** will narrowly focus on Reference Type support, giving time for further generalization with less time constraints.
+
+For usage and scenarios, see [artifact-manifest.md](./artifact-manifest.md)
+
+## Example OCI Artifact Manifests
+
+The following are Phase 1 examples:
+
+- [`net-monitor:v1` oci container image](./artifact-manifest/net-monitor-oci-image.json)
+- [`net-monitor:v1` notary v2 signature](./artifact-manifest/net-monitor-image-signature.json)
+- [`net-monitor:v1` sample sbom](./artifact-manifest/net-monitor-image-sbom.json)
+- [`net-monitor:v1` nydus image with on-demand loading](./artifact-manifest/net-monitor-image-nydus-ondemand-loading.json)
+
+## OCI Artifact Manifest Properties
+
+For **Phase 1**, an artifact manifest provides an optional collection of blobs and a reference to the manifest of another artifact.
+
+- **`schemaVersion`** *int*
+
+  This REQUIRED property specifies the artifact manifest schema version.
+  For this version of the specification, this MUST be `3`. The value of this field WILL change as the manifest schema evolves. Minor version changes to the `oci.artifact.manifest` spec MUST be additive, while major version changes MAY be breaking. Artifact clients MUST implement version checking to allow for future, yet unknown changes. Artifact clients MUST ignore additive properties to minor versions. Artifact clients MAY support major changes, with no guarantee major changes MAY impose breaking changing behaviors. Artifact authors MAY support new and older schemaVersions to provide the best user experience.
+
+- **`mediaType`** *string*
```

Comment:
Embedding the `mediaType` directly in the object itself makes future migration challenging or impossible. I know this supports being able to detect the type of the object, but usually this should be done through a specific descriptor path.

If we want to have this ability, we should come up with a different name. Using the mediaType to set the content of "THIS" document will be confusing, since we use it in both roles.

Reply:
I'm not sure how this correlates with the [^ q&a](https://github.com/opencontainers/artifacts/pull/29#discussion_r674958600).

Reply:
This is a good thread to tug on, for some background: https://github.com/opencontainers/image-spec/pull/411#pullrequestreview-5261352